person.rb

```diff
@@ -0,0 +1,28 @@
+require_relative 'nameable'
+
+class Person < Nameable
+  attr_reader :id, :rentals
+  attr_accessor :name, :age
+
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    super()
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+    @rentals = []
+  end
+
+  def can_use_services?
+    of_age? || @parent_permission
+  end
+
+  def correct_name
+    name
+  end
+
+  private
+
+  def of_age?
+    @age >= 18
+  end
+end
```

Comment:
- [ ] Kindly implement the "add_rental" method which will make sure that the "A person has many rentals" has been implemented correctly.

Reply:
Hi @Nemwel-Boniface 

Thanks for your review.

But regarding the many-to-many relationships between `Book` and `Person` throughout `Rental`, I think there are no methods to implement in `Book` and `Person`. As we initialize a `Rental` instance, that instance is automatically added to the `rentals` instance variable of `book` and `person`.

Moreover, that's what's done in the lesson about `associations`. Please check it out [here](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md).

Cheers,

Reply:
Fixed.